core/src/main/c/freebsd/files.c

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2020 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+#include "../share/files.h"
+#include <sys/mman.h>
+#include <errno.h>
+#include <string.h>
+#include <stdio.h>
+ 
+JNIEXPORT jlong JNICALL JavaCritical_io_questdb_std_Files_mremap0
+        (jlong fd, jlong address, jlong previousLen, jlong newLen, jlong offset, jint flags) {
+    int prot = 0;
+
+    if (flags == com_questdb_std_Files_MAP_RO) {
```

Comment:
perhaps we should extract method body into an inlined function and use the same function in both critical and non-critial JNI wrappers

Reply:
The non critical function is only used by JVM's that dont support the critical path (most do and certainly the openjdk does) and by code running in interpreted mode (according to the docs I read, but this has never happened to me). Certainly in my past experiments with the openjdk when I didn't even have a non critical function prototype it was never called. However, the docs say we need a non critical function prototype...

Reply:
I was trying to make a point about code duplication.  Both function can exist they can just call an inlinable C function